pkg/cmd/roachtest/roachtestutil/clusterupgrade/clusterupgrade.go

```diff
@@ -159,6 +156,16 @@ func UploadWorkload(
 	nodes option.NodeListOption,
 	v *Version,
 ) (string, bool, error) {
+	// minWorkloadBinaryVersion is the minimum version for which we have
+	// `workload` binaries available.
+	var minWorkloadBinaryVersion *Version
+	switch c.Architecture() {
+	case vm.ArchARM64:
+		minWorkloadBinaryVersion = MustParseVersion("v23.2.0")
+	default:
```

Comment:
Nit: explicitly handling `ArchAMD64` and `ArchFIPS` and throwing an error if it's anything else might solve some problems in the future. Doesn't seem to be strictly necessary for now.

Reply:
When did we start publishing FIPS `workload` binaries? Where do I find this information?

Reply:
My understanding is we have never published a FIPS `workload` binary: https://github.com/cockroachdb/cockroach/blob/a4b8a4c214a4ecabddf13d39c4987d3b28fdb187/pkg/cmd/publish-artifacts/main.go#L105

Not sure if I'm missing something.

Reply:
So, really, amending my original comment, `ArchAMD64` and `ArchARM64` would be good, and any other value would be an error. This is fine to address as a follow-up if it seems necesary or helpful.

Reply:
Ah, I believe it's because of this code:

https://github.com/cockroachdb/cockroach/blob/2a311a630c98421eab1f21a9623bb6d7ced0e3fe/pkg/roachprod/roachprod.go#L513-L519

So we never really attempt to download a "FIPS" workload. This is quite confusing.